docs/data-sources/git_repository.md

```diff
@@ -0,0 +1,145 @@
+---
+Title: "Git Repository Data Source"
+Description: |-
+    Creating the Git Repository Data Source.
+---
+
+# Git Repository
+
+The `tanzu-mission-control_git_repository` Data Source allows you to get git repository to a particular scope through Tanzu Mission Control.
+
+[git-repository]: https://docs.vmware.com/en/VMware-Tanzu-Mission-Control/services/tanzumc-using/GUID-26C2D2F3-0E5C-4E56-B875-B7FB003267E4.html
+
+## git-repository Scope
+
+In the Tanzu Mission Control resource hierarchy, there are two levels at which you can specify git repository resources:
+- **object groups** - `cluster_group` block under `scope` sub-resource
+- **Kubernetes objects** - `cluster` block under `scope` sub-resource
+
+**Note:**
+The scope parameter is mandatory in the schema and the user needs to add one of the defined scopes to the script for the provider to function.
+Only one scope per resource is allowed.
+
+## Cluster group scoped Git Repository
+
+### Example Usage
+
+```terraform
+# Read Tanzu Mission Control git repository : fetch cluster group git repository details
+resource "tanzu-mission-control_git_repository" "create_git_repository" {
+  name = "tf-git-repository-name" # Required
+
+  namespace_name = "tf-namespace" #Required
+
+  scope {
+    cluster_group {
+      name = "default" # Required
+    }
+  }
+}
+```
+
+## Cluster scoped Git Repository
+
+### Example Usage
+
+```terraform
+# Read Tanzu Mission Control git repository : fetch cluster git repository details
+resource "tanzu-mission-control_git_repository" "create_git_repository" {
+  name = "tf-git-repository-name" # Required
+
+  namespace_name = "tf-namespace" #Required
+
+  scope {
+    cluster {
+      cluster_name            = "testcluster" # Required
+      provisioner_name        = "attached"    # Default: attached
+      management_cluster_name = "attached"    # Default: attached
+    }
+  }
+}
+```
+<!-- schema generated by tfplugindocs -->
+## Schema
+
+### Required
+
+- `name` (String) Name of the Repository.
+- `namespace_name` (String) Name of Namespace.
+- `scope` (Block List, Min: 1, Max: 1) Scope for the git repository, having one of the valid scopes: cluster, cluster_group. (see [below for nested schema](#nestedblock--scope))
+
+### Optional
+
+- `meta` (Block List, Max: 1) Metadata for the resource (see [below for nested schema](#nestedblock--meta))
+
+### Read-Only
+
+- `id` (String) The ID of this resource.
+- `spec` (List of Object) Spec for the Repository. (see [below for nested schema](#nestedatt--spec))
+- `status` (Map of String) Status for the Repository.
+
+<a id="nestedblock--scope"></a>
+### Nested Schema for `scope`
+
+Optional:
+
+- `cluster` (Block List, Max: 1) The schema for cluster iam policy full name (see [below for nested schema](#nestedblock--scope--cluster))
+- `cluster_group` (Block List, Max: 1) The schema for cluster group iam policy full name (see [below for nested schema](#nestedblock--scope--cluster_group))
```

Comment:
Why is IAM policy coming up in the docs? Can you please make the necessary code changes to ensure this is removed?

Reply:
updated PTAL